### ISSUE TITLE
Error in CI: ERR_PNPM_FROZEN_LOCKFILE_WITH_OUTDATED_LOCKFILE  Cannot perform a frozen installation because the lockfile needs updates

### ISSUE THREAD

**Issue by pinalbhatt** [pnpm/action-setup#40]
My Action Step is 
`
    - name: Install pnpm and dependencies
      uses: pnpm/action-setup@v2.2.1
      with:
        version: 6.x
        run_install: true
`

I am getting below error when running this  action 

Error in CI: ERR_PNPM_FROZEN_LOCKFILE_WITH_OUTDATED_LOCKFILE  Cannot perform a frozen installation because the lockfile needs updates

and when try to fix my lockfile locally with command `fff` i get following error: 
`
npm i --fix-lockfile   
npm ERR! Cannot read properties of null (reading 'matches')

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/un/.npm/_logs/2022-05-01T10_32_11_056Z-debug-0.log
`

**Comment by KSXGitHub** [comment]
> npm i --fix-lockfile

Can you try <code><strong>pnpm</strong> i --fix-lockfile</code>?

**Comment by pinalbhatt** [comment]
@KSXGitHub 
```
pnpm i --fix-lockfile
 ERR_PNPM_SPEC_NOT_SUPPORTED_BY_ANY_RESOLVER  next@12.1.5_react-dom@18.1.0+react@18.1.0 isn't supported by any available resolver.

This error happened while installing the dependencies of next-seo@5.4.0
Progress: resolved 28, reused 29, downloaded 0, added 0
```
Note that ```npm i --legacy-peer-deps``` works fine when using npm.

**Comment by pinalbhatt** [comment]
For now i have fixed issue by adding following to my `.npmrc` file 
```
auto-install-peers=true

strict-peer-dependencies=false
```

**Comment by zkochan** [comment]
> and when try to fix my lockfile locally with command npm i --fix-lockfile i get following error:

Why do you run `npm i` if you use pnpm?

You should run `pnpm install` and commit the lockfile.

**Comment by Innei** [comment]
Is you using latest pnpm the version of 7.x?

**Comment by liuzhaowei55** [comment]
the same error
```shell
➜  toolbox git:(master) pnpm i --fix-lockfile
 ERR_PNPM_SPEC_NOT_SUPPORTED_BY_ANY_RESOLVER  react-hook-form@7.35.0_react@18.2.0 isn't supported by any available resolver.
```

**Comment by liuzhaowei55** [comment]
> the same error
> 
> ```shell
> ➜  toolbox git:(master) pnpm i --fix-lockfile
>  ERR_PNPM_SPEC_NOT_SUPPORTED_BY_ANY_RESOLVER  react-hook-form@7.35.0_react@18.2.0 isn't supported by any available resolver.
> ```

rerun `pnpm install` and commit the lockfile, its ok

**Comment by bartvanremortele** [comment]
Same error here.

```
➜  pnpm i --fix-lockfile
Scope: all 10 workspace projects
.../apps/app:
 ERR_PNPM_SPEC_NOT_SUPPORTED_BY_ANY_RESOLVER  react-hook-form@7.39.4_react@17.0.2 isn't supported by any available resolver.

This error happened while installing the dependencies of @hookform/resolvers@2.9.10
Progress: resolved 309, reused 309, downloaded 0, added 0
```

getting this in CI:

```
Scope: all 10 workspace projects
Lockfile is up to date, resolution step is skipped
 ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up to date with apps/app/package.json

Note that in CI environments this setting is true by default. If you still need to run install in such cases, use "pnpm install --no-frozen-lockfile"
```

**Comment by PaulRBerg** [comment]
Same error! Running `pnpm i --fix-lockfile` does not change the lockfile, but in CI I keep getting an error.

This is with a project that has a peer dependency listed like this:

```json
"peerDependencies": {
  "@ethersproject/bignumber": "5.x"
},
```

**Comment by ramnes** [comment]
You need to update pnpm before running it :)

**Comment by PaulRBerg** [comment]
Yeah, I was on v7.30.1. After upgrading to v8, I can confirm that running `--fix-lockfile` fixed the issue.

**Comment by jonraem** [comment]
Just encountered the same issue in Azure Pipelines. I can also confirm that updating from pnpm 7 to 8 fixed it for me.

**Comment by talik077** [comment]
Same here. Upgraded from pnpm v7 to v8 and fixing the lock file using the `--fix-lockfile` option fixed the issue

**Comment by jyono** [comment]
same problem today. worked for us.

**Comment by echocrow** [comment]
@jyono The spike is probably related to  the [pnpm v8 release](https://github.com/pnpm/pnpm/releases/tag/v8.0.0) just a few hours ago, which seems to require a new lockfile version when installing with `--frozen-lockfile` (which some CI/CD environments default to).

We ran into this issue too, because our CI/CD was installing `pnpm`, pulling the _latest_ version. Until earlier today, that was v7—until that switched to v8, breaking our builds due to the unsupported lockfile version.

Locally installing pnpm v8 and updating the lockfile (as per above posts) worked.
Alternatively, you can also fix your pnpm version to `pnpm@^7` and stick with the old lockfile.

**Comment by PaulRBerg** [comment]
I guess the takeaway from this issue is always to use specific versions of external dependencies in CI.

**Comment by echocrow** [comment]
> I guess the takeaway from this issue is to always use specific versions of external dependencies in CI.

(+1, ironically also why the lockfiles in question exist)

**Comment by njleonzhang** [comment]
pnpm lock is incompatible between v7 and v8? any plan to fix?

**Comment by exsesx** [comment]
I have a `@formkit/auto-animate` dependency, version `1.0.0-beta.6`, pnpm version `8.1.0`

I'm getting the `ERR_PNPM_SPEC_NOT_SUPPORTED_BY_ANY_RESOLVER  @formkit@auto-animate@1.0.0-beta.6 isn't supported by any available resolver.` error in CI.

**Comment by jyono** [comment]
> @jyono The spike is probably related to the [pnpm v8 release](https://github.com/pnpm/pnpm/releases/tag/v8.0.0) just a few hours ago, which seems to require a new lockfile version when installing with `--frozen-lockfile` (which some CI/CD environments default to).
> 
> We ran into this issue too, because our CI/CD was pulling the _latest_ `pnpm` version. Earlier today, that was v7, until this switched to v8, breaking our builds due to the unsupported lockfile version.
> 
> Locally installing pnpm v8 and updating the lockfile (as per above posts) worked.
> 
> Alternatively, you can also fix your pnpm version to `pnpm@^7` and stick with the old lockfile.

thanks!

**Comment by so1ve** [comment]
Summary:
Use corepack to manage package manager is the best choice. This enforces the same manager in different environments. For example, you are running pnpm@7 locally but pnpm@8 in ci, the lockfile format will break the installation. So, please make sure you have `packageManager` field defined in your package.json instead of specifing a pnpm version in the workflow file manually.

**Comment by alex-page** [comment]
If you encounter this error with `pnpm catalog` you might have an incorrectly added `"` or `,` in the `pnpm-workspace.yaml` file.
